File: leedcode/36_Valid_Sudoku.cpp
#include <vector>
#include <map>
#include <iostream>
using namespace std;
bool isValid(vector<vector<char>> &board)
{
	for(int i = 0; i < 9; ++i)
	{
		unordered_map<char,bool> dic1;
		unordered_map<char,bool> dic2;
	    unordered_map<char,bool> dic3;
	    for(int j = 0; j < 9; ++j)
	    {
	    	if(board[i][j] != '.')
	        {
	        	if(dic1[i][j] == true)
	        		return false;
	        	dic1[i][j] == true;
	        }
	        if(board[j][i] != '.')
	        {
	        	if(dic2[i][j] == true)
	        	    return false;
	        	dic2[j][i] == true;
	        }
	        if(board[i/3*3+j/3][i%3*3+j%3] != '.')
	        {
	        	if(dic3[i/3*3+j/3][i%3*3+j%3] == true)
	        		return false;
	        	dic3[i/3*3+j/3][i%3*3+j%3] == true;
	        }
	    }
	}
	return true;
}

int main()
{
	vector<vector<char>> board;
	for(int i = 0; i < 9; ++i)
	{
		for(int j = 0; j < 9; ++j)
		    cin >> board[i][j];
	}
	bool result;
	result = isValid(board);
	return result;
}

File: leedcode/36_Valid_Sudoku.py
def isValid(nums):
	for i in range(9):
		dic1 = {}
	    dic2 = {}
	    dic3 = {}
	    for j in range(9):
	    	if nums[i][j] != '.':
	    		if dic1[nums[i][j]] == True:
	    			return False
	    		dic[nums[i][j]] == True
	    	if nums[j][i] != '.':
	    		if dic2[nums[j][i]] == True:
	    			return False
	    		dic[nums[j][i]] == True
	        if nums[i/3*3+j/3][i%3*3+j%3] != '.':
	        	if dic3[nums[i/3*3+j/3][i%3*3+j%3]] == True:
	        		return False
	        	dic3[nums[i/3*3+j/3][i%3*3+j%3]] = True
	return True


